### PR TITLE
drop support for prometheus 1.x, update prometheus version to 2.52.0

### LINKED ISSUE #718: Remove prometheus 1.x support
While working on fixing the tests for Archlinux I came across a lot of code for prometheus 1.x. Given that prometheus 2.0 was released 2017 it is time to remove that support in order to unclutter the code making it more readable and maintanable.

### PR DESCRIPTION
**Pull request by TheMeier**
also drop support for prometheus  < 2.8.0 (released Mar 12, 2019)

fixes #718

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,100 +44,13 @@ This module automates the install and configuration of Prometheus monitoring too
...
-`manage_prometheus_server` parameter to true in the `prometheus` class. Background information about this change is described in the related [pull request](https://github.com/voxpupuli/puppet-prometheus/pull/187) and the [issue](https://github.com/voxpupuli/puppet-prometheus/issues/184).
-
-For more information regarding class parameters please take a look at the class docstrings.
+* The following exporters are currently implemented: apache_exporter, beanstalkd_exporter, bind_exporter, bird_exporter, blackbox_exporter, collectd_exporter, consul_exporter, dellhw_exporter, elasticsearch_exporter, graphite_exporter, grok_exporter, haproxy_exporter, ipmi_exporter, ipsec_exporter, jmx_exporter, memcached_exporter, mesos_exporter, mongodb_exporter, mysqld_exporter, nginx_prometheus_exporter, nginx_vts_exporter, node_exporter, openldap_exporter, openvpn_exporter, php_fpm_exporter, postfix_exporter, postgres_exporter, process_exporter, puppetdb_exporter, rabbitmq_exporter, redis_exporter, snmp_exporter, ssh_exporter, ssl_exporter, statsd_exporter, systemd_exporter, unbound_exporter, varnish_exporter, wireguard_exporter,
```
Comment (bastelfreak) [inline README.md]: should we just drop it and tell the people to check the REFERENCE.md? That lists all classes/exporters? We will always forget to update the README.md.

File: README.md [hunk 2]
```diff
@@ -398,27 +262,13 @@ prometheus::alertmanager::receivers:
...
-bundle install --path .vendor/ --without system_tests --without development --without release
-bundle exec rake test
-```
-
-## Authors
+See https://voxpupuli.org/docs/how_to_run_tests/ for information on how to run test locally.
 
-puppet-prometheus is maintained by [Vox Pupuli](https://voxpupuli.org), it was written by [brutus333](https://github.com/brutus333/).
+## Transfer Notice
+This plugin was originally authored by it was written by [brutus333](https://github.com/brutus333/)
```
Comment (bastelfreak) [inline README.md]: that sentence is a bit broken.

File: spec/fixtures/files/prometheus2.debian [hunk 3]
```diff
@@ -7,7 +7,7 @@
 # Default-Stop:      S 0 1 6
 # Short-Description: Prometheus monitoring system
 # Description:       The Prometheus monitoring system and time series database. https://prometheus.io/
-#                    
+#
```
Comment (bastelfreak) [inline spec/fixtures/files/prometheus2.debian:10]: is this file even used anymore or can we delete it?

File: README.md (hunk 1 above)
Comment (TheMeier) [reply README.md]: fine with me

File: README.md (hunk 2 above)
Comment (TheMeier) [reply README.md]: oops

File: spec/fixtures/files/prometheus2.debian (hunk 3 above)
Comment (TheMeier) [reply spec/fixtures/files/prometheus2.debian:10]: I wanted to clean up all files in `spec/fixtures/files` in a future PR

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**